app/src/main/java/study_dev/testbed/studyhopper/Group.java

```diff
@@ -0,0 +1,66 @@
+package study_dev.testbed.studyhopper;
+
+import android.widget.ImageView;
+import android.widget.Switch;
+
+public class Group {
+    private String groupName;
+    private String courseCode;
+    private String groupColor;
+    private boolean coedGroup, femalesOnlyGroup, malesOnlyGroup;
+    private int maxGroupMembers;
+    private String documentId;
+
+    public Group() {
+        // empty constructor needed
+    }
+
+    public Group(String groupName, String courseCode, String groupColor,
+                 boolean coedGroup, boolean femalesOnlyGroup, boolean malesOnlyGroup,
+                 int maxGroupMembers) {
+        this.groupName = groupName;
+        this.courseCode = courseCode;
+        this.groupColor = groupColor;
+        this.coedGroup = coedGroup;
+        this.femalesOnlyGroup = femalesOnlyGroup;
```

Comment:
same thing as above, 3 boolean for same value isnt ideal

Reply:
Resolved this issue by modifying the Groups.java class to have only a single string field for groupPreferences.